packages/sdk-core/src/modules/dapp-staking/pending-rewards/index.ts

```diff
@@ -321,7 +322,7 @@ export const getUsedFee = async (
     const blockHash = await api.rpc.chain.getBlockHash(height);
     const signedBlock = await api.rpc.chain.getBlock(blockHash);
     const apiAt = await api.at(signedBlock.block.header.hash);
-    const allRecords = (await apiAt.query.system.events()).toArray();
+    const allRecords: any = await apiAt.query.system.events();
```

Comment:
Please don't use `any` Applicable to multiple locations within the code
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html

Reply:
@bobo-k2 
BTW, Property 'map' does not exist on type 'Codec'.
The type of 'allRecords' is Codec.
I tried to iterate the data by using 'filter' or 'map'. But that make build error(yarn polkadot-exec-tsc --build tsconfig.build.json).

How can I handle this? Can you suggest any reference docs?

Reply:
> @bobo-k2 BTW, Property 'map' does not exist on type 'Codec'. The type of 'allRecords' is Codec. I tried to iterate the data by using 'filter' or 'map'. But that make build error(yarn polkadot-exec-tsc --build tsconfig.build.json).
> 
> How can I handle this? Can you suggest any reference docs?

Codec is a base type from which all polkadot.js types are inherited. `System.events` are of type `Vec<FrameSystemEventRecord>` so you can change your command to
```
const allRecords = await api.query.system.events<Vec<FrameSystemEventRecord>>();
```

Reply:
Codec is a base type from which all polkadot.js types are inherited. `System.events` are of type `Vec<FrameSystemEventRecord>` so you can change your line to
```
const allRecords = await api.query.system.events<Vec<FrameSystemEventRecord>>();
```

Reply:
When '<Vec<FrameSystemEventRecord>>' applied, can not import the member like below error message. 
packages/sdk-core/src/modules/dapp-staking/pending-rewards/index.ts:7:10 - error TS2305: Module '"@polkadot/types/lookup"' has no exported member 'FrameSystemEventRecord'.

Alternative, I used the 'EventRecord' and got properties(actualFee, amount) by index.